test/threads.jl

```diff
@@ -147,3 +147,39 @@ end
 
 # We don't need the watchdog anymore
 close(proc.in)
+
+# https://github.com/JuliaLang/julia/pull/42973
+@testset "spawn and wait *a lot* of tasks in @profile" begin
```

Comment:
```suggestion
Sys.islinux() && @testset "spawn and wait *a lot* of tasks in @profile" begin
```

perhaps limit this to linux, then merge, and open a new PR to remove the conditional and then we can work on any issues found by other platforms?